ethereum/core/src/main/java/org/hyperledger/besu/ethereum/processing/TransactionProcessingResult.java

```diff
@@ -194,6 +196,25 @@ public ValidationResult<TransactionInvalidReason> getValidationResult() {
     return validationResult;
   }
 
+  /**
+   * Set isProcessedInParallel to the value in parameter
+   *
+   * @param isProcessedInParallel new value of isProcessedInParallel
+   */
+  public void setIsProcessedInParallel(final Optional<Boolean> isProcessedInParallel) {
+    this.isProcessedInParallel = isProcessedInParallel;
+  }
```

Comment:
can we avoid to make this class mutable?

Reply:
The issue is that currently, when a transaction executes, it doesn't know itself if it si being executed in parallel or not, so the instance of TransactionProcessingResult is not aware if it was parallel or not. This information is only accessible later on. that's why we need to make the result mutable at this level only.

Reply:
ok I see, fine for me, but maybe this is a sign that some refactor is needed, to postponed the creation of the result, of the data should be kept elsewhere